### DISCUSSION TITLE
func.now() generates a typing error when the column is nullalble

### DISCUSSION THREAD

**Discussion by ramonvg** [sqlalchemy/sqlalchemy#10100] [Usage Questions]
Hello, 

I'm upgrading our codebase to SQLAlchemy 2 and I found this behavior. Assigning `func.now()` to a Mapped[datetime] works successfully but when the column is nullable it generates this mypy error. Is this expected? Thanks in advance.

Code example:
```python
from sqlalchemy.orm import DeclarativeBase
from datetime import datetime
from sqlalchemy.orm import Mapped
from sqlalchemy import func

class Test(DeclarativeBase):
    not_null: Mapped[datetime] 
    nullable: Mapped[datetime | None] 


test = Test()
test.not_null = func.now()
test.nullable = func.now()  # THIS ONE GENERATES THE ERROR
```

Mypy error:
```
> mypy
database/test_datetime.py:13: error: Incompatible types in assignment (expression has type "now", variable has type "SQLCoreOperations[datetime | None] | datetime | None")  [assignment]
```

Versions:
```
> pip freeze | grep -i sqlalchemy
SQLAlchemy==2.0.18

> mypy --version
mypy 1.4.1 (compiled: yes)
```

**Comment by CaselIT** [comment]
Hi,

Sorry missed this one.
Maybe this will be solved by this change #10288 ? cc @zzzeek 

An alternative, that does requires a manual change, is to use `Nullable` added in version 2.0.20 https://docs.sqlalchemy.org/en/20/core/sqlelement.html#sqlalchemy.Nullable

**Comment by zzzeek** [comment]
> Hi,
> 
> Sorry missed this one. Maybe this will be solved by this change #10288 ? cc @zzzeek

wow yeah it should, neat

it's a good test let's just run it

yup it's green with main, with 2.0.20 it fails


this is fixed by #10288 will be in 2.0.21

> An alternative, that does requires a manual change, is to use `Nullable` added in version 2.0.20 https://docs.sqlalchemy.org/en/20/core/sqlelement.html#sqlalchemy.Nullable

neat this works too, that's in 2.0.20

```py
from sqlalchemy import Nullable

test = Test()
test.not_null = func.now()
test.nullable = Nullable(func.now())  # THIS ONE GENERATES THE ERROR
```

**Reply by CaselIT** [reply]
cool! I wasn't sure if `thing` is a considered a subtype of `thing | None` or not. it seems it is here.

**Reply by CaselIT** [reply]
added here 8503dc2e948908199cd8ba4e6b1d1ddcf92f4020
